starlette/responses.py

```diff
@@ -342,5 +379,154 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
                             "more_body": more_body,
                         }
                     )
-        if self.background is not None:
-            await self.background()
+
+    async def _handle_single_range(
+        self, send: Send, start: int, end: int, file_size: int, send_header_only: bool
+    ) -> None:
+        self.headers["content-range"] = f"bytes {start}-{end - 1}/{file_size}"
+        self.headers["content-length"] = str(end - start)
+        await send({"type": "http.response.start", "status": 206, "headers": self.raw_headers})
+        if send_header_only:
+            await send({"type": "http.response.body", "body": b"", "more_body": False})
+        else:
+            async with await anyio.open_file(self.path, mode="rb") as file:
+                await file.seek(start)
+                more_body = True
+                while more_body:
+                    chunk = await file.read(min(self.chunk_size, end - start))
+                    start += len(chunk)
+                    more_body = len(chunk) == self.chunk_size and start < end
+                    await send({"type": "http.response.body", "body": chunk, "more_body": more_body})
+
+    async def _handle_multiple_ranges(
+        self,
+        send: Send,
+        ranges: list[tuple[int, int]],
+        file_size: int,
+        send_header_only: bool,
+    ) -> None:
+        boundary = "".join(random_choices("abcdefghijklmnopqrstuvwxyz0123456789", k=13))
```

Comment:
I'm a bit concerned that this uses predictable randomness

I'd be tempted to go for 19 characters using SystemRandom.choices to get 96 bits of entropy

Reply:
`why join(random_choices)` anyway? 

It's slower than `secrets.token_hex(6)` unless 13 has some special meaning here.

Reply:
Do other frameworks do something different from this?

Would you like to open a PR for it?

Reply:
> 13 has some special meaning here

Nothing special. It's just a random number I picked, and the spec doesn't make any specific requirements on this length.

Reply:
> I'm a bit concerned that this uses predictable randomness
> 
> I'd be tempted to go for 19 characters using SystemRandom.choices to get 96 bits of entropy

This is predicted to be secure...it does not serve any cryptographic purpose.

Reply:
Urllib3 does this https://github.com/urllib3/urllib3/blob/2458bfcd3dacdf6c196e98d077fc6bb02a5fc1df/src/urllib3/filepost.py#L26 (it's for form data but same sort of concept)

nginx is just totally weird https://github.com/nginx/nginx/blob/b1e07409b1071564e8dd8c70436cba5da19a1575/src/core/ngx_file.c#L17 it uses a random number generator based on successive additions of  123456

Reply:
Yes, urllib3 implementation is just equivalent to `secrets.token_hex()`

Reply:
Yeah I'd probably go for secrets.token_hex()